### DISCUSSION TITLE
Default Supplier ID error on part import

### DISCUSSION THREAD

**Discussion by CWatson-Tactile** [inventree/InvenTree#8429] [Q&A]
I am trying to import a list of parts (.xlsx), and I keep getting this error (SupplierPart matching query does not exist.).
It works when I remove the 'Default Supplier ID' column, but I want to have the parts pre-associated with a supplier. 

Is this the wrong field for suppliers?
I already have the suppliers added, and am using the suppliers 'ID #'

**Comment by CWatson-Tactile** [comment]
Forgot to add screen shots
![image](https://github.com/user-attachments/assets/32c18c19-e350-4772-bbc1-f4237fe460d4)
<img width="1248" alt="image" src="https://github.com/user-attachments/assets/4c3c151b-8476-4623-a188-56a61e043d1f">

**Comment by SchrodingersGat** [comment]
Probably what is happening is that the "Supplier" you have created in the system is not actually marked as a supplier. The `is_supplier` field must be set to `True`. Make sure that the supplier is configured correctly and then re-attempt the import

**Comment by CWatson-Tactile** [comment]
All of the suppliers are marked as 'is_supplier = 1'

![Untitled 2](https://github.com/user-attachments/assets/c4f6ca4d-4997-41bb-af41-99e9fdc5d657)

**Reply by SchrodingersGat** [reply]
Ah, ok I see what is happening. The "default supplier" maps to a [Supplier Part](https://docs.inventree.org/en/stable/order/company/#supplier-parts) and not directly to the supplier. So, you need to import the supplier part information and *after that* you can set the "default supplier" field

**Reply by turist79** [reply]
@SchrodingersGat Is it also correct that the link between part and manufacturer`s part for the given stock item only exists "via" supplier`s part record?

**Reply by SchrodingersGat** [reply]
@turist79 not exactly. You can have multiple "manufacturers" for a given part. And also multiple "suppliers" - which may be different to the manufacturer

**Reply by turist79** [reply]
> @turist79 not exactly. You can have multiple "manufacturers" for a given part. And also multiple "suppliers" - which may be different to the manufacturer

yes, sure. but it was referring specifically to the stock item - the possibility to identify stock item as instance of the part produced by one of the manufacturers is only via supplier part record. I mean, for example if i add stock item or edit one, i can only select supplier part (it if exists for that part) linking by this action the stock item to the manufaxcturer part. I can not link the stock item to the manufacturer part directly. Or can i?

**Reply by SchrodingersGat** [reply]
Right. Yes, that's the structure:

`StockItem` -> `SupplierPart` -> `ManufacturerPart`
